### ISSUE TITLE
Build error on ghc 9.2.2

### ISSUE THREAD

**Issue by HugoPeters1024** [well-typed/cborg#295]
Must have been since a recent release as it only occurred after running cabal update beforehand:

```
[ 8 of 14] Compiling Codec.CBOR.Magic ( src/Codec/CBOR/Magic.hs, dist/build/Codec/CBOR/Magic.o, dist/build/Codec/CBOR/Magic.dyn_o )

src/Codec/CBOR/Magic.hs:628:9: error:
    Ambiguous occurrence ‘copyPtrToMutableByteArray’
    It could refer to
       either ‘Prim.copyPtrToMutableByteArray’,
              imported from ‘Data.Primitive.ByteArray’ at src/Codec/CBOR/Magic.hs:112:1-77
           or ‘Codec.CBOR.Magic.copyPtrToMutableByteArray’,
              defined at src/Codec/CBOR/Magic.hs:660:1
    |
628 |         copyPtrToMutableByteArray (ptr `plusPtr` off) mba 0 len
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by FinleyMcIlwaine** [comment]
Duplicates #290 - will be fixed with version 0.2.7.1. In the meantime, constraining `primitive < 0.7.1.0` is a potential workaround.

**Comment by phadej** [comment]
The release would be very much appreciated, as `primitive-0.7.4.0` is the only `primitive` version compatible with GHC-9.4.
